packages/ra-core/src/controller/details/SaveContext.tsx

```diff
@@ -24,7 +27,7 @@ interface SaveContextValue extends SideEffectContextValue {
     saving?: boolean;
 }
 
-export const SaveContext = createContext<SaveContextValue>({});
+export const SaveContext = createContext<SaveContextValue>(undefined);
```

Comment:
I'm surprised TypeScript doesn't complain about the default context value. The type should be `SaveContextValue?` I think.

Reply:
I was surprised as well but yet it doesn't complain about it